awx/api/serializers.py

```diff
@@ -1959,6 +1964,137 @@ def to_representation(self, obj):
         return ret
 
 
+class BulkHostSerializer(HostSerializer):
+    name = serializers.CharField(required=True, allow_blank=False, max_length=512)
+    instance_id = serializers.CharField(required=False, max_length=1024)
+    description = serializers.CharField(required=False)
+    enabled = serializers.BooleanField(default=True, required=False)
+    variables = serializers.CharField(write_only=True, required=False)
+
+    class Meta:
+        fields = (
+            'name',
+            'enabled',
+            'instance_id',
+            'description',
+            'variables',
+        )
```

Comment:
Why don't you specify `model = Host` in the `Meta`? That seems like the most obvious thing to do. I'm not sure if you should even need to declare the fields manually, although maybe subtract `inventory` from the fields.

Reply:
@AlanCoding Sure, I can add the model = Host. For the fields I am following the other serializers, most of them define a model but still have all the fields too. Please let me know if there is a specific use case of defining the fields ?

Reply:
Oh yeah, I was wrong. You need the field list. What the model can buy you is that it'll allow you to avoid _defining serializer fields_. The first level gating criteria is that OPTIONS recognizes the field types. I might check out the branch and see if anything remains that can be deleted.

Reply:
Yeah, so first test:

```json
                "child": {
                    "type": "nested object",
                    "required": true,
                    "read_only": false,
                    "children": {
                        "name": {
                            "type": "string",
                            "required": true,
                            "read_only": false,
                            "label": "Name",
                            "max_length": 512,
                            "help_text": "Name of this host.",
                            "filterable": true
                        },
                        "enabled": {
                            "type": "boolean",
                            "required": false,
                            "read_only": false,
                            "label": "Enabled",
                            "help_text": "Is this host online and available for running jobs?",
                            "filterable": true,
                            "default": true
                        },
                        "instance_id": {
                            "type": "string",
                            "required": false,
                            "read_only": false,
                            "label": "Instance id",
                            "help_text": "The value used by the remote inventory source to uniquely identify the host",
                            "max_length": 1024,
                            "filterable": true,
                            "default": ""
                        },
                        "description": {
                            "type": "string",
                            "required": false,
                            "read_only": false,
                            "label": "Description",
                            "help_text": "Optional description of this host.",
                            "filterable": true,
                            "default": ""
                        },
                        "variables": {
                            "type": "json",
                            "required": false,
                            "read_only": false,
                            "label": "Variables",
                            "help_text": "Host variables in JSON or YAML format.",
                            "filterable": true,
                            "default": ""
                        }
                    }
                },
```

is what I get after doing

```diff
diff --git a/awx/api/serializers.py b/awx/api/serializers.py
index 14f38d004d..b2043025aa 100644
--- a/awx/api/serializers.py
+++ b/awx/api/serializers.py
@@ -1965,12 +1965,6 @@ class GroupSerializer(BaseSerializerWithVariables):
 
 
 class BulkHostSerializer(HostSerializer):
-    name = serializers.CharField(required=True, allow_blank=False, max_length=512)
-    instance_id = serializers.CharField(required=False, max_length=1024)
-    description = serializers.CharField(required=False)
-    enabled = serializers.BooleanField(default=True, required=False)
-    variables = serializers.CharField(write_only=True, required=False)
-
     class Meta:
         model = Host
         fields = (
```

And this is surprisingly _better_ than with the fields in there. Currently, the enabled field gives:

```json
                        "enabled": {
                            "type": "boolean",
                            "required": false,
                            "read_only": false,
                            "label": "Enabled",
                            "filterable": true,
                            "default": true
                        },
```

Because when you specified a DRF field for `enabled` you didn't give help_text, but the model field already has it. So without defining the fields, it'll infer the properties from the Django model fields on `Host`.

Reply:
included in https://github.com/jainnikhil30/awx/pull/16 thanks @AlanCoding